js/cardview.js

```diff
@@ -16,28 +16,120 @@ let Card = React.createClass({
   
   },
 
-  // highlight a card when clicked
-  clickCard: function() {
-    // card can only be highlighted and played if player has enough mana
+  // style the card based on if it has attacked, is castable, etc
+  cssClassesForCard: function(card) {
+    let cssClassCanPlay = '';
     if (this.props.cardInfo.cost > this.props.player.mana) {
+      cssClassCanPlay = "too-expensive";
+    }
+    let cssClass = this.props.cardInfo.hasFocus ? 
+                   'playing-card active-card' : 'playing-card';
+    let cssClassAttacked = this.props.cardInfo.hasAttacked ? 
+                           'has-attacked-card' : '';
+    let cssClassJustPlayed = this.props.cardInfo.enteredPlayThisTurn ? 
+                             'just-played-card' : '';
+    let combinedCSS = cssClass + ' ' + 
+                      cssClassCanPlay + ' ' + 
+                      cssClassAttacked + ' ' + 
+                      cssClassJustPlayed; 
+    return combinedCSS;
+  },
+
+  // click cards in current player's hand or board
+  clickCard: function() {
+    
+    // ATTACK
+    // can click opponent's in play cards
+    // after player has clicked his own in play card to attack
+    var hasActiveCard = false;
```

Comment:
you can use "let" instead of var basically in all these places. the place where it's actually the most useful is in for loops - a "var" in a for loop conflicts with the scope outside the for loop. a "let" does not.